backend/commons/src/test/java/org/sonarsource/sonarlint/core/commons/validation/RegexpValidatorTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarLint Core - Commons
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.commons.validation;
+
+import java.util.Map;
+import java.util.regex.PatternSyntaxException;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegexpValidatorTest {
+
+  public static final String TEST_REGEXP = "[0-9]+";
+
+  @Test
+  void should_throw_exception_on_invalid_regexp() {
+    assertThrows(PatternSyntaxException.class,
+      () -> new RegexpValidator("[4-[8)"));
+  }
+
+  @Test
+  void should_return_empty_invalid_fields() {
+    RegexpValidator validator = new RegexpValidator(TEST_REGEXP);
+
+    InvalidFields invalidFields = validator.validateAll(Map.of(
+      "field1", "12345",
+      "field2", "455668",
+      "field3", "0"
+    ));
+
+    assertFalse(invalidFields.hasInvalidFields());
+    assertThat(invalidFields.getNames()).isEmpty();
```

Comment:
Elaborating on my comment above: mixing vanilla JUnit assertions with AssertJ makes the assertions harder to read, I would suggest to use AssertJ everywhere.